PetaPoco/Core/PocoData.cs

```diff
@@ -124,8 +124,9 @@ public static PocoData ForType(Type type, IMapper defaultMapper)
         {
             if (type == typeof(System.Dynamic.ExpandoObject))
                 throw new InvalidOperationException("Cannot use dynamic types with this method");
-
-            return _pocoDatas.GetOrAdd(type, () => new PocoData(type, defaultMapper));
+            
+            var key = new Tuple<IMapper, Type>(Mappers.GetMapper(type, defaultMapper), type);
+            return _pocoDatas.GetOrAdd(key, () => new PocoData(type, defaultMapper));
```

Comment:
I meant to use the type of the mapper, not the mapper itself (per your comment about keeping the instances around).

```csharp
var mapper = Mappers.GetMapper(type, defaultMapper);
var key = new Tuple<Type, Type>(mapper.GetType(), type);
```

Reply:
Awe shucks! Sorry. Will fix it quick.

Reply:
I presume you want the mapper variable also? Didn't pay attention to that part until after the push.

Reply:
It's just a question of readability. The single line you have is correct, but it's got a lot going on -- and most of the words are either `mapper` or `type`! Two lines makes it clearer what's going on. 

Fixing something like this is a good place for amending your commit. After you make the change, do:

```
# replaces your previous commit with a new one
git commit --amend --no-edit

git push --force
```

It doesn't matter so much, because all of your small commits will get squashed into one before merging, just a chance to try out some git tricks.

Reply:
(Note that in general, while you can do whatever you want to local commits, it's not good practice to "rewrite history" by force pushing changed commits, since it can cause problems for anyone using your commits. In a case like this, though, where your PR is being reviewed and the minor change has been requested, there's no harm.)